src/catch2/meson.build

```diff
@@ -132,133 +109,155 @@ internal_headers = [
   'internal/catch_random_seed_generation.cpp',
   'internal/catch_reporter_registry.cpp',
   'internal/catch_reporter_spec_parser.cpp',
   'internal/catch_result_type.cpp',
+  'internal/catch_reusable_string_stream.cpp',
   'internal/catch_run_context.cpp',
   'internal/catch_section.cpp',
-  'internal/catch_stdstreams.cpp',
-  'catch_session.cpp',
   'internal/catch_singletons.cpp',
-  'internal/catch_reusable_string_stream.cpp',
-  'internal/catch_stringref.cpp',
+  'internal/catch_source_line_info.cpp',
+  'internal/catch_startup_exception_registry.cpp',
+  'internal/catch_stdstreams.cpp',
   'internal/catch_string_manip.cpp',
+  'internal/catch_stringref.cpp',
   'internal/catch_tag_alias_registry.cpp',
-  'catch_test_case_info.cpp',
+  'internal/catch_test_case_info_hasher.cpp',
   'internal/catch_test_case_registry_impl.cpp',
   'internal/catch_test_case_tracker.cpp',
   'internal/catch_test_registry.cpp',
-  'internal/catch_textflow.cpp',
-  'catch_test_spec.cpp',
   'internal/catch_test_spec_parser.cpp',
-  'catch_timer.cpp',
-  'catch_tostring.cpp',
-  'catch_totals.cpp',
-  'catch_version.cpp',
+  'internal/catch_textflow.cpp',
+  'internal/catch_uncaught_exceptions.cpp',
   'internal/catch_wildcard_pattern.cpp',
   'internal/catch_xmlwriter.cpp',
-  'internal/catch_test_case_info_hasher.cpp',
-  'generators/catch_generators_random.cpp',
-  'generators/catch_generator_exception.cpp',
-  'generators/catch_generators.cpp',
   'matchers/catch_matchers.cpp',
   'matchers/catch_matchers_container_properties.cpp',
   'matchers/catch_matchers_exception.cpp',
+  'matchers/catch_matchers_floating_point.cpp',
   'matchers/catch_matchers_predicate.cpp',
+  'matchers/catch_matchers_quantifiers.cpp',
+  'matchers/catch_matchers_string.cpp',
+  'matchers/catch_matchers_templated.cpp',
   'matchers/internal/catch_matchers_impl.cpp',
-  'catch_tag_alias_autoregistrar.cpp',
+  'catch_approx.cpp',
+  'catch_assertion_result.cpp',
+  'catch_config.cpp',
   'catch_get_random_seed.cpp',
-  'internal/catch_decomposer.cpp',
-  'internal/catch_errno_guard.cpp',
-  'internal/catch_lazy_expr.cpp',
-  'internal/catch_leak_detector.cpp',
-  'internal/catch_message_info.cpp',
-  'internal/catch_polyfills.cpp',
-  'internal/catch_startup_exception_registry.cpp',
-  'internal/catch_uncaught_exceptions.cpp',
-  'interfaces/catch_interfaces_capture.cpp',
-  'interfaces/catch_interfaces_config.cpp',
-  'interfaces/catch_interfaces_exception.cpp',
-  'interfaces/catch_interfaces_registry_hub.cpp',
-  'interfaces/catch_interfaces_reporter_factory.cpp',
-  'interfaces/catch_interfaces_reporter_registry.cpp',
-  'interfaces/catch_interfaces_testcase.cpp',
-]
+  'catch_message.cpp',
+  'catch_registry_hub.cpp',
+  'catch_session.cpp',
+  'catch_tag_alias_autoregistrar.cpp',
+  'catch_test_case_info.cpp',
+  'catch_test_spec.cpp',
+  'catch_timer.cpp',
+  'catch_tostring.cpp',
+  'catch_totals.cpp',
+  'catch_version.cpp',
+)
 
 reporter_headers = [
```

Comment:
Why are some changed to `files` and some are not?

Reply:
I mentioned it in the commit message. split() is used later on. That only works for strings, not files.